### DISCUSSION TITLE
Images not rendering in PDF

### DISCUSSION THREAD

**Discussion by hyphen81** [spatie/browsershot#785] [Q&A] [answered]
This works when rendering the view in the browser, but gives me the broken image icon when I generate a PDF:

```
<img src="{{ public_path('images/my-image.png') }}" alt="" style="max-height:3rem;">
```

I've tried hard-coding the path, I've tried using `asset()` instead of public path (which works when rendering in the browser) I've tried using jpg or svg as an alternative, all with the same result.

**Comment by hyphen81** [comment, marked as answer]
in case anyone else runs into this, the answer was to base64 encode the image:
```
            $imagePath = public_path('images/my-image.png');
            $imageType = pathinfo($imagePath, PATHINFO_EXTENSION);
            $imageData = file_get_contents($imagePath);
            $data['base64'] = 'data:image/' . $imageType . ';base64,' . base64_encode($imageData);
```

and then in my blade file:
```	
			<img src="{{ $data['base64'] }}" alt="" style="max-height:3rem;">

```

**Reply by plweil** [reply]
I can confirm that this works. I'm just curious as to why images render locally, but need (at least sometimes) base64 encoding on the server.

**Reply by rimialfa** [reply]
It works for me too.
The reason why local assets are not working is by default, resources from the local network are blocked to prevent Server-Side Request Forgery (SSRF), if you use external image it works.

I think there is a pull request to fix this issue

**Comment by nbyloff** [comment]
I was hoping to avoid the base encoding because it's increasing the file size so much. I couldn't find a PR to address this. Publicly resolvable full path images should work right? It renders just fine in a browser, but not when running through browsershot. I have only gotten the base64 embedded image to show up.

I tried a handful of settings too:

```
--no-sandbox
--allow-file-access-from-files
--disable-web-security
--allow-running-insecure-content
--ignore-certificate-errors
```
